### PR TITLE
#282 - Use dependency groups instead of dev only extras

### PR DESCRIPTION
**Pull request by da1910**
This PR replaces the test and doc extras with dependency groups.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -82,47 +87,30 @@ linux-kerberos = [
...
-  "sphinx_autodoc_typehints",
-  "sphinx-notfound-page",
-  "sphinx-copybutton"
-]
-
 [tool.tox]
 legacy_tox_ini = """
 [tox]
-envlist = py37,py38,py39,py310
+envlist = py37,py38,py39,py310,py311
```
Comment (Andy-Grigg) [inline pyproject.toml:93]: We should do this in BoM Analytics as well if we haven't already.

**Review by Andy-Grigg — APPROVED** [review]
👍

File: pyproject.toml (hunk 1 above)
Comment (da1910) [reply pyproject.toml:93]: It's already done there, only need to turn it on in the strategy matrix and it should test!

**Comment by Andy-Grigg** [comment]
https://github.com/pyansys/openapi-common/issues/282

**Pull request merged**